anta/runner.py

```diff
@@ -49,14 +49,18 @@ async def main(
     coros = []
     for device, test in itertools.product(inventory.get_inventory(established_only=established_only, tags=tags).values(), tests):
         test_class = test[0]
-        test_inputs = test[1]
-        try:
-            # Instantiate AntaTest object
-            test_instance = test_class(device=device, inputs=test_inputs)
-            coros.append(test_instance.test(eos_data=None))
-        except Exception as e:  # pylint: disable=broad-exception-caught
-            message = "Error when creating ANTA tests"
-            anta_log_exception(e, message, logger)
+        test_inputs = test[1] or {}
+        test_tags = test_inputs.get("tags") if "tags" in test_inputs.keys() else ["all"]
+        if tags is None:
+            tags = ["all"]
+        if any(t in test_tags for t in tags):  # type: ignore
+            try:
+                # Instantiate AntaTest object
+                test_instance = test_class(device=device, inputs=test_inputs)
+                coros.append(test_instance.test(eos_data=None))
+            except Exception as e:  # pylint: disable=broad-exception-caught
+                message = "Error when creating ANTA tests"
+                anta_log_exception(e, message, logger)
```

Comment:
I think this logic should not be in the module `anta.runner`.
Instead, it should be somewhere in the `AntaTest` constructor, after the input validation (you want to validate the inputs which contains the tags).
It will be a bit different in the report since you will skip the test. But IMHO it makes more sense (you can even write a clear message on why the test has been skipped with the tags values of the device and the test).

Reply:
could end up with a loooot of skipped result for this reason - probably better to either add a some result state or to never add the test to the ResultManager

Reply:
Exactly. Also, runner is in charge of loading tests and execute them, so it seems better to not load tests we don't want to run with anta